manager/src/main/java/org/openremote/manager/rules/JsonRulesBuilder.java

```diff
@@ -791,15 +793,47 @@ protected RuleActionExecution buildRuleActionExecution(JsonRule rule, RuleAction
             boolean isPush = Objects.equals(notification.getMessage().getType(), PushNotificationMessage.TYPE);
             boolean isHtml;
 
-            if (isEmail) {
+            if (isLocalized) {
+                LocalizedNotificationMessage localizedMsg = (LocalizedNotificationMessage) notification.getMessage();
+                isHtml = false;
+                localizedMsg.getMessages().forEach((lang, msg) -> {
+                    if (msg instanceof PushNotificationMessage pushMsg) {
+                        PushNotificationAction action = pushMsg.getAction();
+                        if (action != null && action.getUrl() != null) {
+                            String newUrl = replaceAssetIdPlaceholder(action.getUrl(), ruleState, useUnmatched, "notification URL", true);
+                            action.setUrl(newUrl);
+                            pushMsg.setAction(action);
+                        }
+                        
+                        if (pushMsg.getBody() != null) {
+                            String newBody = replaceAssetIdPlaceholder(pushMsg.getBody(), ruleState, useUnmatched, "notification body", false);
+                            pushMsg.setBody(newBody);
+                        }
+                    }
```

Comment:
AFAIK it's also possible that the `LocalizedNotificationMessage` can contain an `EmailNotificationMessage` instead of a `PushNotificationMessage`? Would be nice if support for this is also added in the future.